libs/parallelism/execution/include/hpx/execution/algorithms/bulk.hpp

```diff
@@ -115,6 +71,63 @@ namespace hpx { namespace execution { namespace experimental {
                     bulk_receiver<Receiver, Shape, F>(
                         std::forward<Receiver>(receiver), shape, f));
             }
+
+            template <typename Receiver, typename Shape_, typename F_>
+            struct bulk_receiver
+            {
+                std::decay_t<Receiver> receiver;
+                std::decay_t<Shape_> shape;
+                std::decay_t<F_> f;
+
+                template <typename Receiver_, typename Shape__, typename F__>
+                bulk_receiver(Receiver_&& receiver, Shape__&& shape, F__&& f)
+                  : receiver(std::forward<Receiver_>(receiver))
+                  , shape(std::forward<Shape__>(shape))
+                  , f(std::forward<F__>(f))
+                {
+                }
+
+                template <typename E>
+                    void set_error(E&& e) && noexcept
+                {
+                    hpx::execution::experimental::set_error(
+                        std::move(receiver), std::forward<E>(e));
+                }
+
+                void set_done() && noexcept
+                {
+                    hpx::execution::experimental::set_done(std::move(receiver));
+                }
+
+                // The typedef is duplicated from parent struct as the parent one is
+                // not instantiated early enough to use it here.
+                using value_type =
+                    typename hpx::execution::experimental::sender_traits<
+                        Sender>::template value_types<hpx::tuple, hpx::variant>;
+
+                template <typename... Ts>
+                    auto set_value(Ts&&... ts) &&
+                    noexcept -> decltype(
+                        std::declval<hpx::variant<hpx::monostate, value_type>>()
+                            .template emplace<value_type>(
+                                hpx::make_tuple<>(std::forward<Ts>(ts)...)),
+                        void())
```

Comment:
What is the rationale of this change?

Reply:
Some configurations were not tested with C++17, this is needed to sfinae out set_value for `gcc-oldest`